src/main/java/org/ibp/api/java/impl/middleware/ontology/validator/VariableValidator.java

```diff
@@ -532,14 +542,26 @@ private Integer parseVariableTypeAsInteger(final VariableType variableType) {
 	}
 
 	private boolean isAnalysisVariable(final VariableDetails variable) {
-        for(final org.ibp.api.domain.ontology.VariableType type : variable.getVariableTypes()){
-            if(Objects.equals(StringUtil.parseInt(type.getId(), 0), org.generationcp.middleware.domain.ontology.VariableType.ANALYSIS.getId())){
-                return true;
-            }
-        }
-        return false;
+		return this.isVariable(variable, org.generationcp.middleware.domain.ontology.VariableType.ANALYSIS.getId());
     }
 
+	private boolean isGermplasmAttributeVariable(final VariableDetails variable) {
+		return this.isVariable(variable, org.generationcp.middleware.domain.ontology.VariableType.GERMPLASM_ATTRIBUTE.getId());
+	}
+
+	private boolean isGermplasmPassportVariable(final VariableDetails variable) {
+		return this.isVariable(variable, org.generationcp.middleware.domain.ontology.VariableType.GERMPLASM_PASSPORT.getId());
+	}
+
+	private boolean isVariable(final VariableDetails variable, final Integer variableTypeId) {
```

Comment:
@cuenyad can be implemented in VariableDetails with name hasVariableType 
Replace code using anyMatch

Reply:
done,